test/service/test_manifest.py

```diff
@@ -340,16 +423,29 @@ def test_manifest_zarr(self):
             with self.subTest(is_single_part=single_part):
                 with mock.patch.object(type(config), 'disable_multipart_manifests', single_part):
                     response = self._get_manifest(ManifestFormat.compact, filters)
-                    self.assertEqual(200, response.status_code, 'Unable to download manifest')
+                    self.assertEqual(200, response.status_code)
                     # Cannot use response.iter_lines() because of https://github.com/psf/requests/issues/3980
-                    lines = response.content.decode('utf-8').splitlines()
+                    lines = response.content.decode().splitlines()
                     tsv_file = csv.DictReader(lines, delimiter='\t')
                     rows = list(tsv_file)
-                    self.assertEqual(len(rows), 13)  # 12 related file, one original
-                    self.assertEqual(len(rows), len({row['file_uuid'] for row in rows}), 'Rows are not unique')
-                    # 2 pairs of files out of the 13 have matching checksums
-                    self.assertEqual(11, len({row['file_sha256'] for row in rows}), 'Incorrect unique sha256 count')
-                    self.assertEqual(11, len({row['file_crc32c'] for row in rows}), 'Incorrect unique crc32c count')
+                    rows = [dict(file_crc32c=row['file_crc32c'],
+                                 file_name=row['file_name'],
+                                 file_uuid=row['file_uuid'],
+                                 specimen_from_organism_organ=row['specimen_from_organism.organ']) for row in rows]
+                    self.assertEqual(expected, rows)
+
+                response = self._get_manifest(ManifestFormat.curl, filters)
+                self.assertEqual(200, response.status_code)
+                lines = response.content.decode().splitlines()
+                file_prefix = 'output="587d74b4-1075-4bbf-b96a-4d1ede0481b2/'
+                curl_files = []
+                for line in lines:
+                    if line.startswith(file_prefix) and line.endswith('"'):
```

Comment:
If a line starts with `file_prefix` it MUST end in `"`. Your current implementation would still fail the assertion below if it came across a line lacking the trailing quotation mark. That's acceptable, but the failure wouldn't be very informative since it conflates the absence of a line with a malformed line.

Reply:
Actually, the current implementation is wrong. If the response contained the same line twice, once with and once without trailing quotation marks, the test would pass.